informers.go

```diff
@@ -23,14 +34,89 @@ func getVirtCli() (kubecli.KubevirtClient, error) {
 	return virtCli, err
 }
 
-func getVmiInformer() (cache.SharedIndexInformer, error) {
-	virtCli, err := getVirtCli()
-	if err != nil {
-		return nil, err
-	}
-
+func getVmiInformer(virtCli kubecli.KubevirtClient) (cache.SharedIndexInformer, error) {
 	listWatcher := cache.NewListWatchFromClient(virtCli.RestClient(), "virtualmachineinstances", k8sv1.NamespaceAll, fields.Everything())
 	vmiInformer := cache.NewSharedIndexInformer(listWatcher, &k6tv1.VirtualMachineInstance{}, 1*time.Hour, cache.Indexers{})
+	
+	vmiInformer.AddEventHandler(cache.ResourceEventHandlerFuncs {
+		DeleteFunc: removeWarningLabel,
```

Comment:
I'm not entirely sure, but I think that reboots might not delete the vmi entirely, but rather leave in a `Succeeded` state. Just to be on the safe side, I'd also add an update handler and re-use the same function you already have.

I would also add a small validation to `removeWarningLabel()` to make sure the VMI is either deleted or in a non-running phase.

```suggestion
		DeleteFunc: removeWarningLabel,
		UpdateFunc: func(_, newObj interface{}) {
			removeWarningLabel(newObj)
		},
```